### PR TITLE
Add deployment cancel callback block

### PR DESCRIPTION
**Pull request by YiMysty**
Add deployment cancel callback block
Tested thru review-lab https://cancel-test.review-lab.github.com/paper-spa/minyi

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -122,4 +125,30 @@ async function main() {
...
+            'Content-type': 'application/json'
+          }
+        }
+      )
+      core.info(`canceled ongoing deployment thru ${pagesCancelDeployEndpoint}`)
+    }
+  } catch (e) {
+    console.info('cancel deployment errored', e)
+  }
+  process.exit(isNaN(+evtOrExitCodeOrError) ? 1 : +evtOrExitCodeOrError)
```
Comment (tcbyrd) [inline src/index.js:147]: 💯

**Review by tcbyrd — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -122,4 +125,30 @@ async function main() {
   }
 }
 
+async function cancelHandler(evtOrExitCodeOrError) {
+  try {
+    if (requestedDeployment) {
+      const pagesCancelDeployEndpoint = `https://api.github.com/repos/${context.repositoryNwo}/pages/deployment/cancel/${context.buildVersion}`
```
Comment (jcudit) [inline src/index.js:131]: Seeing `https://api.github.com/repos/${context.repositoryNwo}/pages/` used in a few places.  Would be more readable to declare it early and reuse it as needed.

**Review by jcudit — APPROVED** [review]

File: src/index.js (hunk 2 above)
Comment (YiMysty) [reply src/index.js:131]: Will adjust in separate pr~

Comment (tcbyrd) [reply src/index.js:131]: Yeh it's not optimal. Eventually this endpoint will become part of the `github` package anyway.

**Pull request merged**